avm/res/hybrid-container-service/provisioned-cluster-instance/main.bicep

```
@@ -0,0 +1,361 @@
+metadata name = 'Hybrid Container Service Provisioned Cluster Instance'
+metadata description = 'Deploy a provisioned cluster instance.'
+
+@description('Required. The name of the provisioned cluster instance.')
+param name string
+
+@description('Optional. Location for all Resources.')
+param location string = resourceGroup().location
+
+@description('Optional. Tags of the resource.')
+param tags object?
+
+@description('Optional. Enable/Disable usage telemetry for module.')
+param enableTelemetry bool = true
+
+@description('Optional. The name of the secret in the key vault that contains the SSH private key PEM.')
+param sshPrivateKeyPemSecretName string = 'AksArcAgentSshPrivateKeyPem'
+
+@description('Optional. The name of the secret in the key vault that contains the SSH public key.')
+param sshPublicKeySecretName string = 'AksArcAgentSshPublicKey'
+
+@description('Conditional. The name of the key vault. The key vault name. Required if no existing SSH keys.')
+param keyVaultName string?
+
+@description('Required. The id of the Custom location that used to create hybrid aks.')
+param customLocationId string
```

```suggestion
param customLocationResourecId string
```

To align with [SNFR22](https://azure.github.io/Azure-Verified-Modules/spec/SNFR22/#id-snfr22---category-inputs---parametersvariables-for-resource-ids). Technically the same would apply to the vnetSubnetId that is part of the `cloudProviderProfileType` UDT - but I don't want to overcomplicate it for you on the first version. This may need to be changed when moving to 1.0 though